rpm/tarantool.spec

```diff
@@ -159,6 +167,9 @@ C and Lua/C modules.
          -DWITH_SYSTEMD:BOOL=ON \
          -DSYSTEMD_UNIT_DIR:PATH=%{_unitdir} \
          -DSYSTEMD_TMPFILES_DIR:PATH=%{_tmpfilesdir} \
+%endif
+%if 0%{?fedora} >= 33
+         -DENABLE_LTO=ON \
```

Comment:
In fact, we set our own flags `-flto -fno-fat-lto-objects` (via `/usr/share/cmake/Modules/Compiler/GNU.cmake`) (and `-Wno-lto-type-mismatch` for a linker) after [ones](https://fedoraproject.org/wiki/LTOByDefault) from Fedora 33 (`%_lto_cflags` == `-flto=auto -ffat-lto-objects`). Fedora 35 [will](https://fedoraproject.org/wiki/Changes/LTOBuildImprovements) enable `-fno-fat-lto-objects` by default, because there is no much sense to store compiled code in object/archive files if you don't install them (and it saves some build time). According to GCC [docs](https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html) `-flto` should make attempt to detect GNU Make job server (as well as `-flto=auto`) if I understood the following wording right:

> You can also specify `-flto=jobserver` to use GNU make’s job server mode to determine the number of parallel jobs. \<...\> Even without the option value, GCC tries to automatically detect a running GNU make’s job server.

So, to sum up, the change looks okay for me. We don't discard anything important from the distribution specific build options.